qiime2/core/util.py

```diff
@@ -116,14 +118,32 @@ def duration_time(relative_delta):
         return '0 %s' % attrs[-1]
 
 
+def has_md5sum_native():
+    return shutil.which('md5sum') is not None
+
+
 def md5sum(filepath):
+    if has_md5sum_native():
+        return md5sum_native(filepath)
+    else:
+        return md5sum_python(filepath)
+
+
+def md5sum_python(filepath):
     md5 = hashlib.md5()
     with open(str(filepath), mode='rb') as fh:
         for chunk in iter(lambda: fh.read(io.DEFAULT_BUFFER_SIZE), b""):
             md5.update(chunk)
     return md5.hexdigest()
 
 
+def md5sum_native(filepath):
+    result = subprocess.run(['md5sum', str(filepath)],
+                            check=True, capture_output=True, text=True)
+    _, digest = from_checksum_format(result.stdout)
+    return digest
+
+
```

Comment:
Why did this whole deal need to happen?

Reply:
Using the binary at all? -> for speed
Detecting whether it's present? -> not a universally present program (I don't have it e.g.)

Reply:
Speed mostly, it's not super important as we're in the same order of magnitude for running md5sum, but it could save up to an hour on larger imports.